src/repr/src/adt/mz_acl_item.rs

```diff
@@ -637,3 +782,145 @@ fn test_mz_acl_item_binary() {
+    assert!(!acl.acl_mode.contains(AclMode::SELECT));
+    assert!(!acl.acl_mode.contains(AclMode::UPDATE));
+    assert!(!acl.acl_mode.contains(AclMode::DELETE));
+    assert!(!acl.acl_mode.contains(AclMode::USAGE));
+    assert!(!acl.acl_mode.contains(AclMode::CREATE));
+    assert!(!acl.acl_mode.contains(AclMode::CREATE_ROLE));
+    assert!(!acl.acl_mode.contains(AclMode::CREATE_DB));
+    assert!(!acl.acl_mode.contains(AclMode::CREATE_CLUSTER));
+    assert_eq!(s, acl.to_string());
+
+    let s = "1=RBN/2";
+    let acl: AclItem = s.parse().unwrap();
+    assert_eq!(1, acl.grantee.0);
+    assert_eq!(2, acl.grantor.0);
+    assert!(!acl.acl_mode.contains(AclMode::INSERT));
+    assert!(!acl.acl_mode.contains(AclMode::SELECT));
+    assert!(!acl.acl_mode.contains(AclMode::UPDATE));
+    assert!(!acl.acl_mode.contains(AclMode::DELETE));
+    assert!(!acl.acl_mode.contains(AclMode::USAGE));
+    assert!(!acl.acl_mode.contains(AclMode::CREATE));
+    assert!(acl.acl_mode.contains(AclMode::CREATE_ROLE));
+    assert!(acl.acl_mode.contains(AclMode::CREATE_DB));
+    assert!(acl.acl_mode.contains(AclMode::CREATE_CLUSTER));
+    assert_eq!(s, acl.to_string());
+
+    assert!("42/rw=666".parse::<AclItem>().is_err());
+    assert!("u42=rw/u666".parse::<AclItem>().is_err());
+    assert!("s42=rw/s666".parse::<AclItem>().is_err());
+    assert!("u32=C/".parse::<AclItem>().is_err());
+    assert!("=/".parse::<AclItem>().is_err());
+    assert!("f62hfiuew827fhh".parse::<AclItem>().is_err());
+    assert!("u2=rw/s66=CU/u33".parse::<AclItem>().is_err());
+}
+
+#[mz_ore::test]
+fn test_acl_item_binary() {
+    use std::ops::BitAnd;
+
+    let acl_item = AclItem {
+        grantee: Oid(42),
+        grantor: Oid(666),
+        acl_mode: AclMode::empty()
+            .bitand(AclMode::SELECT)
+            .bitand(AclMode::UPDATE),
+    };
+    assert_eq!(
+        acl_item,
+        AclItem::decode_binary(&acl_item.encode_binary()).unwrap()
+    );
+
+    let acl_item = AclItem {
+        grantee: PUBLIC_ROLE_OID,
+        grantor: Oid(4),
+        acl_mode: AclMode::empty()
+            .bitand(AclMode::USAGE)
+            .bitand(AclMode::CREATE),
+    };
+    assert_eq!(
+        acl_item,
+        AclItem::decode_binary(&acl_item.encode_binary()).unwrap()
+    );
+
+    let acl_item = AclItem {
+        grantee: Oid(7),
+        grantor: Oid(12),
+        acl_mode: AclMode::empty(),
+    };
+    assert_eq!(
+        acl_item,
+        AclItem::decode_binary(&acl_item.encode_binary()).unwrap()
+    );
+
+    let acl_item = AclItem {
+        grantee: PUBLIC_ROLE_OID,
+        grantor: Oid(100),
+        acl_mode: AclMode::empty(),
+    };
+    assert_eq!(
+        acl_item,
+        AclItem::decode_binary(&acl_item.encode_binary()).unwrap()
+    );
+
+    assert!(AclItem::decode_binary(&[1, 2, 3, 4, 5, 6, 7, 8, 9, 0]).is_err())
+}
+
+#[mz_ore::test]
+fn test_acl_item_binary_size() {
+    assert_eq!(16, AclItem::binary_size());
+}
```

Comment:
Can you add some `proptest`s here? For example,
```
proptest! {
  #[test]
  fn proptest_acl_item_binary_encoding_roundtrip(a: AclItem) {
    // encode a
    // decode a and assert
  }

  #[test]
  fn proptest_valid_acl_item_str(s: "[\d]?=[\w]{0-3}\/[\d]{0-3}") {
    // parse s
  }
}
```
Note: proptest can take a regex and create strings matching it. I'm not sure if the regex I provided is exactly what we need so definitely tweak it if need be!